MaterialLib/FractureModels/MohrCoulomb.cpp

```diff
@@ -82,22 +81,34 @@ void MohrCoulomb<DisplacementDim>::computeConstitutiveRelation(
     }
 
     // check shear yield function (Fs)
-    double const Fs = std::abs(sigma[0]) + sigma[1] * std::tan(mat.phi) - mat.c;
+    double const Fs = std::abs(tau) + sigma_n * std::tan(mat.phi) - mat.c;
+
     material_state_variables.setShearYieldFunctionValue(Fs);
     if (Fs < .0)
     {
         Kep = Ke;
         return;
     }
 
-    Eigen::VectorXd dFs_dS(2);
-    dFs_dS[0] = boost::math::sign(sigma[0]);
-    dFs_dS[1] = std::tan(mat.phi);
-
+    Eigen::VectorXd dFs_dS(DisplacementDim);
     // plastic potential function: Qs = |tau| + Sn * tan da
-    Eigen::VectorXd dQs_dS(2);
-    dQs_dS[0] = boost::math::sign(sigma[0]);
-    dQs_dS[1] = std::tan(mat.psi);
+    Eigen::VectorXd dQs_dS(DisplacementDim);
+
+    if (DisplacementDim == 2)
+    {
+        dFs_dS[0] = boost::math::sign(tau);
+        dQs_dS[0] = boost::math::sign(tau);
+    }
+    else
+    {
+        for (int i=0; i<index_ns-1; i++)
+            dFs_dS[i] = boost::math::sign(tau) / tau * sigma[i];
```

Comment:
Here `sigma[i] / std::abs(tau)` produces less assembly, and I believe it is equivalent. In case tau = 0 the expression is not defined anyway...